File: 백준/Silver/1065. 한수/한수.py
def check(i):
    if (len(str(i))<=2):
        return True
    
    diff = int(str(i)[1]) - int(str(i)[0])
    
    for index in range (1,len(str(i))):
        if (diff != int(str(i)[index]) - int(str(i)[index-1])):
            return False
    return True
        
N = int(input())
cnt = 0

for i in range(1,N+1):
    if check(i):
       cnt+=1

print(cnt)

File: 백준/Silver/1065. 한수/한수.kt
import java.util.Scanner

fun check(i: Int): Boolean {
    if (i < 100) return true // 100미만의 수는 모두 한수

    val numStr = i.toString()
    val diff = numStr[1] - numStr[0]
    
    for (index in 1 until numStr.length) {
        if (diff != numStr[index] - numStr[index - 1]) {
            return false
        }
    }
    return true
}

fun main() {
    val scanner = Scanner(System.`in`)
    val N = scanner.nextInt()
    var cnt = 0

    for (i in 1..N) {
        if (check(i)) {
            cnt++
        }
    }

    println(cnt)
}